src/Microsoft.AspNetCore.Razor.Language/DefaultTagHelperDescriptorBuilder.cs

```diff
@@ -32,9 +29,41 @@ public DefaultTagHelperDescriptorBuilder(string kind, string name, string assemb
             _metadata = new Dictionary<string, string>(StringComparer.Ordinal);
         }
 
+        public override string DisplayName { get; set; }
+
+        public override ICollection<string> AllowedChildTags
+        {
+            get
+            {
+                if (_allowedChildTags == null)
+                {
+                    _allowedChildTags = new HashSet<string>();
+                }
+
+                return _allowedChildTags;
+            }
+        }
+
+        public override string TagOutputHint { get; set; }
+
+        public override string Documentation { get; set; }
+
         public override IDictionary<string, string> Metadata => _metadata;
 
-        public override TagHelperDescriptorBuilder BindAttribute(Action<BoundAttributeDescriptorBuilder> configure)
+        public override ICollection<RazorDiagnostic> Diagnostics
+        {
+            get
+            {
+                if (_diagnostics == null)
+                {
+                    _diagnostics = new HashSet<RazorDiagnostic>();
+                }
+
+                return _diagnostics;
+            }
+        }
+
+        public override void BindAttribute(Action<BoundAttributeDescriptorBuilder> configure)
```

Comment:
@NTaylorMullen, I know we discussed exposing the underlying list as a readonly collection, but looking at how this turned out I felt like there was not much value in having that here. Plus there was nothing exposing it prior to this change. Am I missing something?